src/RaceChrono.cpp

```diff
@@ -72,7 +130,34 @@ void RaceChronoBleAgent::setUp(
 
   Bluefruit.setTxPower(+4);
 }
+#elif defined(ARDUINO_ARCH_ESP32)
+void RaceChronoBleAgent::setUp(
+    const char *bluetoothName, RaceChronoBleCanHandler *handler) {
+  ::handler = handler;
+  BTName = bluetoothName;
+  BLEDevice::init(BTName);
+  BLEDevice::setSecurityIOCap(BLE_HS_IO_NO_INPUT_OUTPUT);
+  pServer = BLEDevice::createServer();
+  pMainService = pServer->createService(MAIN_SERVICE_NAME);
+
+  _pidRequestsCharacteristic = pMainService->createCharacteristic( 
+                PID_CHARACTERISTIC, 
+                NIMBLE_PROPERTY::WRITE
+                );
+  _pidRequestsCharacteristic->setCallbacks(&cbhandle_racechrono_filter_request);
+
+  _canBusDataCharacteristic = pMainService->createCharacteristic( 
+                CANBUS_CHARACTERISTIC, 
+                NIMBLE_PROPERTY::READ |
+                NIMBLE_PROPERTY::NOTIFY
+                );
+  pMainService->start();
+  
+}
+#endif
 
+
+#if defined(ARDUINO_ARCH_NRF52)
 void RaceChronoBleAgent::startAdvertising() {
```

Comment:
I'd suggest keeping the definitions of common functions/methods outside the #if/#else/etc, and instead move those into the body of the functions/methods. This way both builds are guaranteed to have the same "prototype" for each class, and only the definitions will differ...

Which makes me think...

We should just define an abstract parent class for RaceChronoBleAgent, and provide two device-specific implementations. Similar to how arduino-CAN is structured. What do you think?

Reply:
I think your idea would be a cleaner way to do it.  I don't really like having the entire file a sequence of alternating `#ifdef`/`#else` directives.  Readability suffers greatly.  The parent class can hold the common code mentioned in a comment above, while the device-specific classes being.. well.. device-specific.
Would you like me to do that refactoring in this PR?

Reply:
Yeah, I think it's better for everyone to refactor before we merge.
Others are not in a rush to get this support, and looks like you can make use of it as-is without waiting to finish refactoring.
Thanks!!